drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -98,6 +98,19 @@
 
 #define MMCSD_CAPACITY(b, s)    ((s) >= 10 ? (b) << ((s) - 10) : (b) >> (10 - (s)))
 
+#define MMCSD_USLEEP(usec) \
+  do \
+  { \
+    if (up_interrupt_context()) \
+      { \
+        up_udelay(usec); \
```

Comment:
why not call `up_udelay()` in `nxsig_usleep()` if `up_interrupt_context()` ?

Reply:
In general why wait in interrupt context. That's the whole flaw here, to simply put it you can't wait inside an interrupt.

Reply:
@PetervdPerk-NXP Agree with you that such an implementation should not be used in the interrupt context. I think the changes here might be due to some reasons, such as coredump (dumping data via MMC/SD SDIO devices in interrupt context)? 

Beyond this issue, I am wondering if it is necessary to make `nxsig_usleep()` interrupt-safe. This would help some developers use similar interfaces without worrying too much about the internal implementation.

Reply:
I would rather suggest to add an assertion that triggers if `nxsig_usleep` is called in an interrupt context.

Reply:
> why not call `up_udelay()` in `nxsig_usleep()` if `up_interrupt_context()` ?

OK,I will move call to `nxsig_usleep`

Reply:
No, it shouldn't nxsig_sleep is called by nanosleep, too POSIX is already defined the correct behaviour, we shouldn't change it.

Reply:
So the correct approach is to remove the call to nxsig_sleep() and use the non-signal version, right?
I think there is no good sleep function in the driver module now, and some key operations are not suitable to be interrupted by signals.

Reply:
we can create an interal api for driver, but it's a new topic, not related to this patch directly, and better to address in the new patch.